### ISSUE TITLE
[BUG] Some tests of `limit_test` fail on Spark 3.4

### ISSUE THREAD

**Issue by HaoYang670** [NVIDIA/spark-rapids#7122]
**Describe the bug**
```
FAILED ../../src/main/python/limit_test.py::test_non_zero_offset_with_limit[0-0]
FAILED ../../src/main/python/limit_test.py::test_non_zero_offset_with_limit[0-10]
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.
```
./run_pyspark_from_build.sh -k limit_test
```

**Expected behavior**

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**

**Comment by HaoYang670** [comment]
maybe related to https://issues.apache.org/jira/browse/SPARK-39873

**Comment by HaoYang670** [comment]
Not sure whether this is a bug introduced by https://github.com/NVIDIA/spark-rapids/pull/5946

**Comment by HaoYang670** [comment]
cc @firestarman

**Comment by jlowe** [comment]
Fixed by #7170
